pkg/export/otel/common.go

```diff
@@ -333,24 +336,63 @@ func (l *LogrAdaptor) WithName(name string) logr.LogSink {
 	return &LogrAdaptor{inner: l.inner.With("name", name)}
 }
 
-// headersFromEnv returns a map of the headers as specified by the
-// OTEL_EXPORTER_OTLP_*HEADERS group of variables. This is,
-// a comma-separated list of key=values. For example:
-// api-key=key,other-config-value=value
 func headersFromEnv(varName string) map[string]string {
-	headersStr, ok := os.LookupEnv(varName)
+	headers := map[string]string{}
+
+	addToMap := func(k string, v string) {
+		headers[k] = v
+	}
+
+	parseOTELEnvVar(varName, addToMap)
+
+	return headers
+}
+
+type sinkFunc func(k string, v string)
+
+// parseOTELEnvVar parses a comma separated group of variables
+// in the format specified by OTEL_EXPORTER_OTLP_*HEADERS or
+// OTEL_RESOURCE_ATTRIBUTES, i.e. a comma-separated list of
+// key=values. For example: api-key=key,other-config-value=value
+// The values are passed as parameters to the sink function sFunc
+func parseOTELEnvVar(varName string, sFunc sinkFunc) {
```

Comment:
I elected to pass `sinkFunc` instead of returning `map[string]string` so that

1. we don't have to further convert from a map to `[]attribute.KeyValue` in the `OTEL_RESOURCE_ATTRIBUTES` case
2. enables the caller to filter out and transform/discard any key=value pairs if push comes to shove (I was doing that in an intermediate version but it was no longer required, still nothing to lose here).

Reply:
Maybe I'm missing context, but I don't understand the meaning of `sinkFunc`. Maybe we could chose a more descriptive name?

Reply:
Context: https://en.wikipedia.org/wiki/Sink_(computing) - TLDR: basically a function that receives data for further processing.

I've renamed it to `varHandler` (in the context of OTEL vars) - but happy to come with a less crappy name.